xrpl4j-client/src/main/java/org/xrpl/xrpl4j/client/XrplClient.java

```diff
@@ -252,6 +260,180 @@ public FeeResult fee() throws JsonRpcClientErrorException {
+          .filter("tesSUCCESS"::equals)
+          .map($ -> true)
+          .isPresent();
+
+        final boolean metadataExists = transactionResult.metadata().isPresent();
+
+        if (isTesSuccess) {
+          LOGGER.debug("Transaction with hash: {} was validated with success", transactionHash);
+          return Finality.builder()
+            .finalityStatus(FinalityStatus.VALIDATED_SUCCESS)
+            .resultCode(transactionResult.metadata().get().transactionResult())
+            .build();
+        } else if (!metadataExists) {
+          return Finality.builder().finalityStatus(FinalityStatus.VALIDATED_UNKNOWN).build();
+        } else {
+          LOGGER.debug("Transaction with hash: {} was validated with failure", transactionHash);
+          return Finality.builder()
+            .finalityStatus(FinalityStatus.VALIDATED_FAILURE)
+            .resultCode(transactionResult.metadata().get().transactionResult())
+            .build();
+        }
+      }).orElseGet(() -> {
+        try {
+          final boolean isTransactionExpired = FluentCompareTo.is(getMostRecentlyValidatedLedgerIndex())
+            .greaterThan(lastLedgerSequence);
+          if (!isTransactionExpired) {
+            LOGGER.debug("Transaction with hash: {} has not expired yet, check again", transactionHash);
+            return Finality.builder().finalityStatus(FinalityStatus.NOT_FINAL).build();
+          } else {
+            boolean isMissingLedgers = ledgerGapsExistBetween(UnsignedLong.valueOf(submittedOnLedgerIndex.toString()),
+              UnsignedLong.valueOf(lastLedgerSequence.toString()));
+            if (isMissingLedgers) {
+              LOGGER.debug("Transaction with hash: {} has expired and rippled is missing some to confirm if it" +
+                " was validated", transactionHash);
+              return Finality.builder().finalityStatus(FinalityStatus.NOT_FINAL).build();
+            } else {
+              AccountInfoResult accountInfoResult = this.accountInfo(
+                AccountInfoRequestParams.of(account)
+              );
+              UnsignedInteger accountSequence = accountInfoResult.accountData().sequence();
+              if (FluentCompareTo.is(sequence).lessThan(accountSequence)) {
+                // a different transaction with this sequence has a final outcome.
+                // this represents an unexpected case
+                throw new IllegalStateException("Something unexpected happened. Tx not final.");
+              } else {
+                LOGGER.debug("Transaction with hash: {} has expired, consider resubmitting with updated" +
+                  " lastledgersequence and fee", transactionHash);
+                return Finality.builder().finalityStatus(FinalityStatus.EXPIRED).build();
+              }
+            }
+          }
+        } catch (JsonRpcClientErrorException e) {
+          LOGGER.warn(e.getMessage(), e);
+          return Finality.builder().finalityStatus(FinalityStatus.NOT_FINAL).build();
```

Comment:
This scenario is not covered by the unit tests covering this method (i.e., `isFinal`)